src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -267,5 +317,24 @@ private bool IsNetCoreSDKSmallerThanTheMinimumVersion(string netcoreSdkVersion,
         }
 
 
+        class ResolverLogger
```

Comment:
I had the environment variable in the version where it was writing log files to the temp folder.  In this version, you don't need to opt in, whenever there's a resolver failure the details will be included in MSBuild log.

Reply:
The current implementation keeps a list of messages to log and doesn't do anything with them if everything succeeds.  Should we be worried about the scenario of a large solution loading 100+ projects with this resolver called 100+ times?  In Visual Studio since projects are evaluated in parallel, the caching doesn't work as well and this SDK resolver is called a lot more than we thought.  In this case, a bunch of strings are allocated and then thrown away, just wondering if since SDK resolvers are on the "hot path" if they need to be cognizant of allocations and overall perf.

Reply:
Yes, I think we do need to worry about the perf here.

In this implementation, I don't think any strings are actually allocated unless there is a failure.  There is the list of messages which is allocated, and then each method is either a constant string or a FormattableString with references to constant or already-allocated strings.

So I believe that allocation is kept to a minimum here.

Is there a better way to check the perf than to check the change in and see if the VS perf tests catch a regression?

Reply:
If SDK-style projects are used in DDRITs/RPS then I would expect that to flag any perf degradation I suppose.  I'm not sure if only legacy projects are used though.

Reply:
Could you (manually) capture a dump before and after this change after failing to resolve an SDK?